src/js/containers/search/filters/AgencyListContainer.jsx

```diff
@@ -0,0 +1,205 @@
+/**
+* AgenciesListContainer.jsx
+* Created by Emily Gullo 12/23/2016
+**/
+
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+
+import Autocomplete from 'components/sharedComponents/autocomplete/Autocomplete';
+
+import * as SearchHelper from 'helpers/searchHelper';
+import * as agencyActions from 'redux/actions/search/agencyActions';
+
+const propTypes = {
+    setAutocompleteAwardingAgencies: React.PropTypes.func,
+    setAutocompleteFundingAgencies: React.PropTypes.func,
+    fundingAgencies: React.PropTypes.array,
+    awardingAgencies: React.PropTypes.array,
+    selectAgency: React.PropTypes.func,
+    selectedAgencies: React.PropTypes.object,
+    agencyType: React.PropTypes.string
+};
+
+export class AgencyListContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+
+        this.state = {
+            agencySearchString: '',
+            autocompleteAgencies: []
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.timeout = null;
+    }
+
+    componentDidMount() {
+        if (this.props.agencyType === 'Funding') {
+            this.parseAutocompleteAgencies(this.props.fundingAgencies);
+        }
+        else {
+            this.parseAutocompleteAgencies(this.props.awardingAgencies);
+        }
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.agencyType === 'Funding') {
+            this.parseAutocompleteAgencies(nextProps.fundingAgencies);
+        }
+        else if (this.props.agencyType === 'Awarding') {
+            this.parseAutocompleteAgencies(nextProps.awardingAgencies);
+        }
+    }
+
+    parseAutocompleteAgencies(results) {
+        const agencies = [];
+
+        // Format results of search for use in Autocomplete component
+        if (results && results.length > 0) {
+            results.forEach((item) => {
+                // Push two items to the autocomplete entries if subtier = toptier
+                if (item.toptier_agency.name === item.subtier_agency.name) {
+                    agencies.push({
+                        title: item.subtier_agency.name,
+                        data: Object.assign({}, item, {
+                            agencyType: 'toptier'
+                        })
+                    });
+                }
+
+                agencies.push({
+                    title: item.subtier_agency.name,
+                    subtitle: `Sub-Agency of ${item.toptier_agency.name}`,
+                    data: Object.assign({}, item, {
+                        agencyType: 'subtier'
+                    })
+                });
+            });
+
+            // Remove any selected Toptier Agencies
+            const toptierSelections = _.filter(
```

Comment:
Can we combine these `_.filter` and `_.remove` into a single `_.remove` using a more complex [predicate](https://lodash.com/docs/4.17.4#remove)?